### PR TITLE
add webview to install new JDK

### PR DESCRIPTION
**Pull request by Eskibear**
For this webview, there are some differences in implementation:
* use webview-ui-toolkit for consistent styling with vscode core.
* use inline html referencing a bundled .js file (a `index.js` instead of `index.html`)

### CODE REVIEW

**Comment by Eskibear** [comment]
Demo: 

https://user-images.githubusercontent.com/2351748/141050430-a72bf843-54dd-4960-a4ad-f1e99ee9bc3d.mp4

File: src/install-jdk/asset/app/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
```
Comment (jdneo) [inline src/install-jdk/asset/app/hooks.ts:1]: license (same for other files)

File: src/install-jdk/asset/App.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+class App extends React.Component {
+  render() {
+    return (
+      <InstallJDKView />
+    );
+  }
+}
+
+export default App;
```
Comment (jdneo) [inline src/install-jdk/asset/App.tsx:16]: new empty line (same for other files)

File: src/install-jdk/asset/features/InstallJDKView.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                </PanelView>
+                <PanelView id="view-2">
+                    <OtherJDKsPanel />
+                </PanelView>
+            </Panels>
+            <div className="footer">
+                <p>After you finish JDK installation, please reload Visual Studio Code to make it effective.</p>
+                <div>
+                    <Button appearance='secondary' onClick={onWillReloadWindow}>Reload Window</Button>
+                    <Link href={encodeExternalLinkWithTelemetry(WEBVIEW_ID, "Having trouble?", helpLink)}>Having trouble?</Link>
```
Comment (jdneo) [inline src/install-jdk/asset/features/InstallJDKView.tsx:41]: Add some margin space between `Reload Window` & `having Trouble?`

File: src/install-jdk/index.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+class InstallJdkPage {
+	public static instance: InstallJdkPage | undefined;
+	private static readonly viewType = WEBVIEW_ID;
+	private readonly _panel: vscode.WebviewPanel;
+	private readonly _extensionPath: string;
+	private _disposables: vscode.Disposable[] = [];
+
+	public static createOrShow(extensionPath: string, webviewPanel?: vscode.WebviewPanel) {
+		// "smart" Beside
+		const column = vscode.window.activeTextEditor?.viewColumn !== vscode.ViewColumn.One ?
```
Comment (jdneo) [inline src/install-jdk/index.ts:29]: Seems that viewColumn will be undefined when the active editor is a webview

Comment (Eskibear) [reply src/install-jdk/index.ts:29]: Yes, e.g. VS Code's own getting started page.     Do you have any idea to also cover these cases?

BTW, I don't like `ViewColumn.Beside` as it always creates a new Column, which is wired when I already have multiple columns.

File: src/install-jdk/asset/features/InstallJDKView.tsx (hunk 3 above)
Comment (Eskibear) [reply src/install-jdk/asset/features/InstallJDKView.tsx:41]: Agree. Will update.

File: src/install-jdk/index.ts (hunk 4 above)
Comment (jdneo) [reply src/install-jdk/index.ts:29]: Have no idea how to solve this 🤔 . Maybe use `Beside` when it's undefined?

**Comment by jdneo** [comment]
I got error `Error: Webview is disposed` when

1. Open runtime page
2. Open download page
3. Close download page
4. Close runtime page

File: src/install-jdk/index.ts (hunk 4 above)
Comment (Eskibear) [reply src/install-jdk/index.ts:29]: worth a try.

File: src/install-jdk/constants.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export const WEBVIEW_ID = "java.installJdk";
```
Comment (jdneo) [inline src/install-jdk/constants.ts:1]: license

Comment (Eskibear) [reply src/install-jdk/constants.ts:4]: added

**Review by jdneo — APPROVED** [review]

**Pull request merged**